src/lib/tp.c

```diff
@@ -0,0 +1,152 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2018 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Authors:
+ *         Otavio Pontes <otavio.pontes@intel.com>
+ *
+ */
+
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "macros.h"
+#include "tp.h"
+
+/*
```

Comment:
According to the "how to contribute guide" (https://github.com/clearlinux/swupd-client/pull/757/files#diff-6026cca73952788765f35267bc5da11c) you are not supposed to leave empty lines in the comments

Reply:
I'm starting to rethink about that. Don't you think that

    /*
     * This function does this.
     * bla bla bla
     */

is easier to read than

    /* This function does this.
     * bla bla bla */

Reply:
I'm fine with either format